upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go

```diff
@@ -149,27 +155,47 @@ func (e *Elastigroup) Find(c *fi.Context) (*Elastigroup, error) {
 
 		// Image.
 		{
-			image, err := resolveImage(cloud, fi.StringValue(lc.ImageID))
-			if err != nil {
-				return nil, err
+			if e.ImageID != nil && lc.ImageID != nil &&
```

Comment:
It looks like we won't set `actual.ImageID` in e.g. the case when e.ImageId == lc.ImageID.  I took a look at how "raw AWS" does it - it sets `actual.ImageID = lc.ImageID`.  Then it tries to avoid spurious changes by doing this logic.  But you do need to continue to set ImageID in the common case.